modin/core/dataframe/algebra/binary.py

```diff
@@ -195,6 +196,7 @@ def caller(
                 Result of binary function.
             """
             axis = kwargs.get("axis", 0)
+            shape_hint = None
             if isinstance(other, type(query_compiler)):
```

Comment:
let's retrieve columns at the beginning so the further if-statements be shorter
```suggestion
            shape_hint = None
            self_cols = query_compiler._modin_frame._columns_cache
            if isinstance(other, type(query_compiler)):
                other_cols = other._modin_frame._columns_cache
```

Reply:
fixed